Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -251,39 +255,40 @@ extension InternalRouter where Self: Router {
         
         calculateRoutes(from: location, along: routeProgress) { [weak self] (session, result) in
             guard let self = self else { return }
-            self.isRerouting = false
-            
+
             guard case let .success(indexedResponse) = result else {
-                return
+                self.isRerouting = false; return
             }
             let response = indexedResponse.routeResponse
-            guard let route = response.routes?.first else { return }
+            guard let route = response.routes?.first else {
+                self.isRerouting = false; return
+            }
             
             self.lastProactiveRerouteDate = nil
             
             guard let firstLeg = route.legs.first, let firstStep = firstLeg.steps.first else {
-                return
+                self.isRerouting = false; return
             }
             
             let routeIsFaster = firstStep.expectedTravelTime >= RouteControllerMediumAlertInterval &&
                 currentUpcomingManeuver == firstLeg.steps[1] && route.expectedTravelTime <= 0.9 * durationRemaining
             
-            if routeIsFaster {
-                var routeOptions: RouteOptions?
-                if case let .route(options) = response.options {
-                    routeOptions = options
-                }
-                
-                // Prefer the most optimal route (the first one) over the route that matched the original choice.
-                self.indexedRouteResponse = .init(routeResponse: response, routeIndex: 0)
-                
-                // If the upcoming maneuver in the new route is the same as the current upcoming maneuver, don’t announce it.
-                // FIXME: There must be a better way to skip a redundant initial instruction than to assume the old route’s spoken instruction index is appliable to the new route.
```

Comment:
It doesn't look like this code works with new NavNative implementation. We need to find another way of skipping these instructions on the NavNative side.